### PR TITLE
Eliminate testutils

### PR DESCRIPTION
**Pull request by bazsi**
I had a lot of libtest related changes in my typing PR #3831. I couldn't help but notice that it is only a few users of testutils left, so I went ahead and eliminated testutils completely. 

The branch contains some of the improvements I've made in the parallel branch, simply because I needed them in new test code, I left it here, but it might mean that there are no users of a specific function.

While extracting of the patches I had a few conflicts that I needed to resolve, I hope this is a clean state.

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build FAILURE

**Review by MrAnno — DISMISSED** [review]

**Comment by MrAnno** [comment]
The CMake build is failing because we did not allow including `libtest/...` from itself.

bazsi/syslog-ng#12 should fix it.

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Review by MrAnno — APPROVED** [review]

**Comment by MrAnno** [comment]
2 internal repositories were affected, we've ported ~30 tests to Criterion.
I think we are ready now :)

**Review by gaborznagy — APPROVED** [review]
I've only one minor comment, besides that it looks good to me.

File: libtest/msg_parse_lib.c [hunk 1]
```diff
@@ -78,7 +78,8 @@ assert_log_message_value(LogMessage *self, NVHandle handle, const gchar *expecte
   gssize key_name_length;
   gssize value_length;
   const gchar *key_name = log_msg_get_value_name(handle, &key_name_length);
-  const gchar *actual_value = log_msg_get_value(self, handle, &value_length);
+  const gchar *actual_value_r = log_msg_get_value(self, handle, &value_length);
+  gchar *actual_value = g_strndup(actual_value_r, value_length);
```
[the remark this answers is not in the document]
Comment (MrAnno) [reply libtest/msg_parse_lib.c:82]: Let's fix that test leak separately, so we don't block the "typing" PRs even further.

**Pull request merged**